docs/App/Header.js

```diff
@@ -117,10 +117,11 @@ class Header extends Component<HeaderProps, HeaderState> {
   componentDidMount() {
     this.getStarCount();
   }
-  UNSAFE_componentWillReceiveProps({ location }: HeaderProps) {
+  componentDidUpdate(prevProps: HeaderProps) {
     const valid = ['/', '/home'];
-    const shouldCollapse = !valid.includes(this.props.location.pathname);
-    if (location.pathname !== this.props.location.pathname && shouldCollapse) {
+    const shouldCollapse = !valid.includes(prevProps.location.pathname);
+
+    if (prevProps.location.pathname !== this.props.location.pathname && shouldCollapse) {
```

Comment:
Doing this on update means that the first time you click away from the home page, the transition won't work (which is actually true of the status quo as well).

Instead I would suggest that we call `this.setState({ contentHeight: this.content.scrollHeight });` immediately in `componentDidMount` and get rid of `toggleCollapse` altogether.

Reply:
You're right. I removed `toggleCollapse` alltogether and removed `contentHeight` from the `state` itself. My suggested approach will get rid of one additional re-render after `componentDidMount` and the functionality remains the same.

Reply:
@Methuselah96 I tested this locally and doesn't seem to impact anything negatively. Any other concerns or can this conversation be resolved?